rotkehlchen/globaldb/handler.py

```diff
@@ -1011,6 +1011,123 @@ def add_historical_prices(entries: List['HistoricalPrice']) -> None:
+            if cursor.fetchone()[0] is None:
+                return
+            cursor.execute(
+                """INSERT OR IGNORE INTO price_history(
+                from_asset, to_asset, source_type, timestamp, price
+                ) VALUES (?, ?, ?, ?, ?)
+                """,
+                serialized,
+            )
+        except sqlite3.IntegrityError as e:
+            connection.rollback()  # roll back any of the executemany that may have gone in
+            log.error(
+                f'One of the given historical price entries caused a DB error. {str(e)}. ',
+            )
+        connection.commit()
+
+    @staticmethod
+    def get_manual_prices(asset: 'Asset') -> List[Dict[str, str]]:
+        """Returns prices added to the database by the user for an asset"""
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        querystr = (
+            'SELECT from_asset, to_asset, source_type, timestamp, price FROM price_history '
+            'WHERE from_asset=? AND source_type="A" ORDER BY timestamp'
+        )
+        query = cursor.execute(querystr, (asset.identifier,))
+        result = query.fetchall()
+        return [
+            {
+                'from_asset': entry[0],
+                'to_asset': entry[1],
+                'timestamp': entry[3],
+                'price': entry[4],
+            }
+            for entry in result
+        ]
+
+    def edit_manual_price(self, entry: HistoricalPrice) -> bool:
+        """Returns prices added to the database by the user for an asset"""
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        # check that the entry we want to edit exitst
+        querystr = (
+            'SELECT COUNT(*) FROM price_history '
+            'WHERE from_asset=? AND to_asset=? AND timestamp=? AND source_type="A"'
+        )
+        params = (
+            entry.from_asset.identifier,
+            entry.to_asset.identifier,
+            entry.timestamp,
+        )
+        query = cursor.execute(querystr, params)
+        if query.fetchone()[0] == 0:
+            self.add_single_historical_price(entry)
+            return True
+
+        querystr = (
+            'UPDATE price_history SET price=? WHERE from_asset=? AND to_asset=? '
+            'AND timestamp=? AND source_type="A"'
+        )
+        entry_serialized = entry.serialize_for_db()
+        params_update = (
+            entry_serialized[4],
+            entry_serialized[0],
+            entry_serialized[1],
+            entry_serialized[3],
+        )
+        try:
+            cursor.execute(querystr, params_update)
+        except sqlite3.IntegrityError as e:
+            connection.rollback()
+            log.error(
+                f'Failed to edit manual historical prices from {entry.from_asset} '
+                f'to {entry.to_asset} at timestamp: {str(entry.timestamp)} due to {str(e)}',
+            )
+            return False
+        return True
```

Comment:
This logic is too complicated and not needed. Simply do the update. And then check if `Cursor.rowcount` is one. If it is , the update happened. IF not it did not.